lib/redux/actions/async.js

```diff
@@ -463,6 +470,43 @@ export function clickDeviceSelectionDone() {
   };
 }
 
+export function clickEditUserNext(profile) {
+  return (dispatch, getState) => {
+    const { uploadTargetUser } = getState();
+    const { api } = services;
+    dispatch(syncActions.updateProfileRequest());
+    if (!_.isEmpty(profile)){
+      api.user.updateProfile(uploadTargetUser, profile, (err, profile) => {
+        if (err) {
+          dispatch(syncActions.updateProfileFailure(err));
+        } else {
+          dispatch(syncActions.updateProfileSuccess());
+          async.series([
+            api.user.account,
+            api.user.loggedInProfile,
+            api.user.getUploadGroups
+          ], (err, results) => {
+            dispatch(syncActions.setAllUsers(results[0],results[1],results[2]));
+            const { targetDevices, devices, allUsers, loggedInUser } = getState();
+            const targetedDevices = _.get(targetDevices, uploadTargetUser, []);
+            const supportedDeviceKeys = Object.keys(devices);
+            const atLeastOneDeviceSupportedOnSystem = _.some(targetedDevices, (key) => {
+              return _.includes(supportedDeviceKeys, key);
+            });
+            if (_.isEmpty(targetedDevices) || !atLeastOneDeviceSupportedOnSystem) {
+              return dispatch(syncActions.setPage(pages.SETTINGS));
+            } else {
```

Comment:
I think you're right, now that I'm looking at it. I can't remember the reason I had that wrapped in the `if`. If I remember what my logic was, I can add it back in later. I'll take it out for now.